### PR TITLE
Improve ol.Overlay extensibility

### PR DESCRIPTION
**Pull request by gberaudo**
This PR:
- splits updatePixelPosition_() and make it protected;
- introduces a protected setVisible() function;
- makes autoPan_ protected.

This change is a simple refactoring and improve extensibity by using `@protected`.

### CODE REVIEW

File: src/ol/overlay.js [hunk 1]
```diff
@@ -455,27 +455,48 @@ ol.Overlay.prototype.setPositioning = function(positioning) {
...
 
 /**
- * @private
+ * Modify the visibility of the element.
+ * @param {boolean} visible
+ * @protected
  */
-ol.Overlay.prototype.updatePixelPosition_ = function() {
+ol.Overlay.prototype.setVisibility = function(visible) {
+  if (this.rendered_.visible ^ visible) {
```
Comment (tschaub) [inline src/ol/overlay.js]: Since we're not using bitwise operators for this type of comparison elsewhere, I think `!==` would be clearer here.

Comment (gberaudo) [inline src/ol/overlay.js]: OK, commit updated.

File: src/ol/overlay.js [hunk 2]
```diff
@@ -455,27 +455,48 @@ ol.Overlay.prototype.setPositioning = function(positioning) {
...
+ol.Overlay.prototype.setVisible = function(visible) {
+  if (this.rendered_.visible !== visible) {
+    goog.style.setElementShown(this.element_, visible);
+    this.rendered_.visible = visible;
+  }
+};
+
 
+/**
+ * Update pixel position
```
Comment (elemoine) [inline src/ol/overlay.js]: Please add the final punctuation.

**Comment by elemoine** [comment]
Looks good to me. I think this can be merged when the remaining comments are addressed. Thanks @gberaudo.

**Comment by gberaudo** [comment]
Thank you for the review @tschaub and @elemoine.
I addressed all the comments and the PR is now ready for merging.

**Comment by elemoine** [comment]
Thanks.

**Pull request merged**